components/content-service/pkg/service/usagereport-service.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package service
+
+import (
+	"context"
+
+	"github.com/opentracing/opentracing-go"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/gitpod-io/gitpod/common-go/tracing"
+	"github.com/gitpod-io/gitpod/content-service/api"
+	"github.com/gitpod-io/gitpod/content-service/api/config"
+	"github.com/gitpod-io/gitpod/content-service/pkg/storage"
+)
+
+// UsageReportService implements UsageReportServiceServer
+type UsageReportService struct {
+	cfg config.StorageConfig
+	s   storage.PresignedAccess
+
+	api.UnimplementedUsageReportServiceServer
+}
+
+// NewUsageReportService create a new usagereport service
+func NewUsageReportService(cfg config.StorageConfig) (res *UsageReportService, err error) {
+	s, err := storage.NewPresignedAccess(&cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &UsageReportService{cfg: cfg, s: s}, nil
+}
+
+// UploadURL provides a URL to which clients can upload the content via HTTP PUT.
+func (us *UsageReportService) UploadURL(ctx context.Context, req *api.UsageReportUploadURLRequest) (resp *api.UsageReportUploadURLResponse, err error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "UsageReport.UploadURL")
+	span.SetTag("name", req.Name)
+	defer tracing.FinishSpan(span, &err)
+
+	err = us.s.EnsureExists(ctx, req.Bucket)
```

Comment:
Why is the bucket specified in the request? I would've thought that the UsageReportService owns the location and as such should not be passed in from the outside. In this form, it mostly provides an API for abitrary file upload which is not necessarily what we want.

Reply:
Good point, thanks. I've removed the bucket name from the RPC and used a constant instead (ebba8d090)